packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/io.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+final String flutterRootPath = getFlutterRoot();
+final Directory flutterRoot = fileSystem.directory(flutterRootPath);
+
+Future<void> main() async {
+  // Regression test for https://github.com/flutter/flutter/issues/132592
+  test('flutter/bin/dart updates the Dart SDK', () async {
+    // Run the Dart entrypoint once to ensure the Dart SDK is downloaded.
+    await runDartBatch();
+
+    // Remove the Dart SDK stamp to cause the Dart batch entrypoint to
+    // re-download the Dart SDK.
+    expect(dartSdkStamp.existsSync(), true);
+    dartSdkStamp.deleteSync();
+
+    // Run the Dart batch entrypoint again to trigger the Dart SDK update.
+    final stdout = await runDartBatch();
```

Comment:
```
   info • Missing type annotation • packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:29:5 • always_specify_types
```

Reply:
It turns out that developing Dart while also repeatedly clobbering your Dart SDK is hard 😂